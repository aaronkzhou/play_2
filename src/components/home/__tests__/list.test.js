import React from 'react';
import { render } from '@testing-library/react';
import TestRenderer from 'react-test-renderer'

describe('PostList', () => {
    let PostList

    beforeAll(() => {
        PostList = jest.requireActual('../list').default
    })

    it('should render properly with feeding data with small amount data', () => {
        const renderer = TestRenderer.create(
            <PostList posts={[{ id: 1, title: 'title' }]} />
        )
        expect(renderer.toJSON()).toMatchSnapshot()
    })

    it('should render properly with feeding data with a bit more data', () => {
        const renderer = TestRenderer.create(
            <PostList posts={[
                { id: 1, title: 'title' },
                { id: 2, title: 'title' },
                { id: 3, title: 'title' },
                { id: 4, title: 'title' },
                { id: 5, title: 'title' },
                { id: 6, title: 'title' },
                { id: 7, title: 'title' },
                { id: 8, title: 'title' },
                { id: 9, title: 'title' }
            ]}
            />
        )
        expect(renderer.toJSON()).toMatchSnapshot()
    })
})