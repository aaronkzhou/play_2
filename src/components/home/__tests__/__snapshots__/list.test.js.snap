// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostList should render properly with feeding data with a bit more data 1`] = `
Array [
  <ul
    className="list-container"
  >
    <li
      className="list-wrapper"
    >
      <a
        href="/posts/1"
        onClick={[Function]}
      >
        <div
          className="overview-wrapper"
        >
          title
        </div>
      </a>
    </li>
    <li
      className="list-wrapper"
    >
      <a
        href="/posts/2"
        onClick={[Function]}
      >
        <div
          className="overview-wrapper"
        >
          title
        </div>
      </a>
    </li>
    <li
      className="list-wrapper"
    >
      <a
        href="/posts/3"
        onClick={[Function]}
      >
        <div
          className="overview-wrapper"
        >
          title
        </div>
      </a>
    </li>
    <li
      className="list-wrapper"
    >
      <a
        href="/posts/4"
        onClick={[Function]}
      >
        <div
          className="overview-wrapper"
        >
          title
        </div>
      </a>
    </li>
    <li
      className="list-wrapper"
    >
      <a
        href="/posts/5"
        onClick={[Function]}
      >
        <div
          className="overview-wrapper"
        >
          title
        </div>
      </a>
    </li>
    <li
      className="list-wrapper"
    >
      <a
        href="/posts/6"
        onClick={[Function]}
      >
        <div
          className="overview-wrapper"
        >
          title
        </div>
      </a>
    </li>
  </ul>,
  <button
    className="button"
    onClick={[Function]}
  >
    Load More...
  </button>,
]
`;

exports[`PostList should render properly with feeding data with small amount data 1`] = `
<ul
  className="list-container"
>
  <li
    className="list-wrapper"
  >
    <a
      href="/posts/1"
      onClick={[Function]}
    >
      <div
        className="overview-wrapper"
      >
        title
      </div>
    </a>
  </li>
</ul>
`;
